app/src/main/java/com/dmonzonis/nookpendium/MainActivity.kt

```diff
@@ -0,0 +1,151 @@
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import android.widget.CheckBox
+import android.widget.LinearLayout
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.widget.SearchView
+import androidx.fragment.app.Fragment
+import com.google.android.material.navigation.NavigationView
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.filter_menu.*
+import kotlinx.android.synthetic.main.toolbar.*
+
+
+class MainActivity : AppCompatActivity() {
+    private lateinit var sharedPrefs: SharedPreferences
+    lateinit var filterManager: FilterManager
+    lateinit var sortManager: SortManager
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        sharedPrefs = getSharedPreferences(getString(R.string.shared_prefs), Context.MODE_PRIVATE)!!
+
+        navGamesView.setNavigationItemSelectedListener(navListener())
+
+        val critterFragment = CritterFragment(sharedPrefs.getInt("selectedGame", R.string.game_acnh))
+        supportFragmentManager.beginTransaction()
+            .replace(R.id.fragment_container, critterFragment)
+            .commit()
+
+        filterManager = FilterManager(critterFragment)
+        sortManager = SortManager(critterFragment)
+        setUpFilterManager()
+        setUpSortManager()
+
+        toolbar.inflateMenu(R.menu.toolbar_menu)
+        setSupportActionBar(toolbar)
+        supportActionBar!!.apply {
+            setDisplayShowTitleEnabled(false)
+            setDisplayHomeAsUpEnabled(true)
+            setHomeAsUpIndicator(R.drawable.ic_menu_white_24dp)
+        }
+    }
+
+    private fun navListener() : NavigationView.OnNavigationItemSelectedListener {
+        return NavigationView.OnNavigationItemSelectedListener {
+            var selectedFragment : Fragment? = null
+
+            when (it.itemId) {
+                R.id.miGameAcnh -> {
+                    selectedFragment = CritterFragment(R.string.game_acnh)
+                    sharedPrefs.edit().putInt("selectedGame", R.string.game_acnh).apply()
+                }
+                R.id.miGameAcnl -> {
+                    selectedFragment = CritterFragment(R.string.game_acnl)
+                    sharedPrefs.edit().putInt("selectedGame", R.string.game_acnl).apply()
+                }
+                R.id.miAbout -> selectedFragment = AboutFragment()
+            }
+            if (selectedFragment != null) {
+                supportFragmentManager.beginTransaction()
+                    .replace(R.id.fragment_container, selectedFragment).commit()
+                if (selectedFragment is CritterFragment) {
+                    filterManager.fragment = selectedFragment
+                    sortManager.fragment = selectedFragment
```

Comment:
actually now that we have fragments it is a bit awkward to be passing the fragment around into the filter/sort controllers... we should maybe refactor the listeners that are set up within them, and set them up on the fragments instead, calling the appropriate methods from the controller when necessary.

I can refactor that with another commit in this PR if you want, or I can make a separate PR